nixos/modules/system/boot/luksroot.nix

```diff
@@ -254,8 +254,27 @@ let
         for try in $(seq 3); do
             ${optionalString yubikey.twoFactor ''
             echo -n "Enter two-factor passphrase: "
-            read -r k_user
-            echo
+            k_user=
+            while true; do
+                if [ -e /crypt-ramfs/k_user ]; then
+                    echo "reused"
+                    k_user=$(cat /crypt-ramfs/k_user)
+                    break
+                else
+                    # Try reading it from /dev/console with a timeout
+                    IFS= read -t 1 -r k_user
```

Comment:
I know you copy pasted it but isn't 1 second of timeout a bit short ?

Reply:
Thank you for your review!

Increasing it would certainly save a few iterations for many users, but otherwise I’m not sure what difference it makes. Would you advise increasing it for all three scenarios (not only YubiKey but also [passphrase](https://github.com/NixOS/nixpkgs/blob/0ed35bd664b6110d231d139a72a4a0df05c8ccd7/nixos/modules/system/boot/luksroot.nix#L172) and [GPG Card](https://github.com/NixOS/nixpkgs/blob/0ed35bd664b6110d231d139a72a4a0df05c8ccd7/nixos/modules/system/boot/luksroot.nix#L372))?

Reply:
I don't request you to change this code, rather I don't understand how it works today at all in the places from which you copy pasted it. How can the user type the passphrase in less than a second ? There must be something I haven't understood.